sycl/include/sycl/ext/oneapi/experimental/cuda/builtins.hpp

```diff
@@ -25,37 +25,15 @@ namespace oneapi {
 namespace experimental {
 namespace cuda {
 
-namespace detail {
-using ldg_vector_types = sycl::detail::type_list<
-    sycl::vec<char, 2>, sycl::vec<char, 3>, sycl::vec<char, 4>,
-    sycl::vec<signed char, 2>, sycl::vec<signed char, 3>,
-    sycl::vec<signed char, 4>, sycl::vec<short, 2>, sycl::vec<short, 3>,
-    sycl::vec<short, 4>, sycl::vec<int, 2>, sycl::vec<int, 3>,
-    sycl::vec<int, 4>, sycl::vec<long, 2>, sycl::vec<long, 3>,
-    sycl::vec<long, 4>, sycl::vec<long long, 2>, sycl::vec<long long, 3>,
-    sycl::vec<long long, 4>, sycl::vec<unsigned char, 2>,
-    sycl::vec<unsigned char, 3>, sycl::vec<unsigned char, 4>,
-    sycl::vec<unsigned short, 2>, sycl::vec<unsigned short, 3>,
-    sycl::vec<unsigned short, 4>, sycl::vec<unsigned int, 2>,
-    sycl::vec<unsigned int, 3>, sycl::vec<unsigned int, 4>,
-    sycl::vec<unsigned long, 2>, sycl::vec<unsigned long, 3>,
-    sycl::vec<unsigned long, 4>, sycl::vec<unsigned long long, 2>,
-    sycl::vec<unsigned long long, 3>, sycl::vec<unsigned long long, 4>,
-    sycl::vec<half, 2>, sycl::vec<half, 3>, sycl::vec<half, 4>,
-    sycl::vec<float, 2>, sycl::vec<float, 3>, sycl::vec<float, 4>,
-    sycl::vec<double, 2>, sycl::vec<double, 3>, sycl::vec<double, 4>>;
-
-using ldg_types =
-    sycl::detail::tl_append<ldg_vector_types,
-                            sycl::detail::gtl::scalar_floating_list,
-                            sycl::detail::gtl::scalar_signed_integer_list,
-                            sycl::detail::gtl::scalar_unsigned_integer_list>;
-} // namespace detail
-
 template <typename T>
 inline __SYCL_ALWAYS_INLINE std::enable_if_t<
-    sycl::detail::is_contained<
-        T, sycl::ext::oneapi::experimental::cuda::detail::ldg_types>::value,
+    detail::check_type_in_v<detail::element_type_t<T>, char, signed char, short,
```

Comment:
I think this part isn't NFC in a sense that `gtl::scalar_floating_list` on the left included `bfloat16` but the code in this hasn't been adjusted to actually handle it when it was added there. As such, the change here is actually good.

Reply:
Should we leave a TODO reminding it on code?

Reply:
IMO, that's up to the @intel/llvm-reviewers-cuda to decide if any changes are necessary here.